### ISSUE TITLE
AWS SES - not able to respond to mail because of outgoing email address randomization

### ISSUE THREAD

**Issue by Addvilz** [simple-login/app#1513]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report

**Describe the bug**
SES uses a seemingly pseudo-random address as both mail from and return path.

For example, if a user has a domain `example.org` in SES and inbox `user@example.org`, for all outgoing mail the mail from and return path can be either (depending on configuration):

1. `XXX@<region>.amazonses.com`; OR
2. `XXX@YYY.example.org`

Where XXX is a pseudo-random issued by SES, also used as Message-ID, and unique per each mail.

Thus it can never really be deterministic and hence there seems no real way to be able to respond to mail forwarded by SimpleLogin as long as SES is used as mail transport (authorized addresses is not an option).

One potential fix would be to potentially allow whitelist Authorized domains, or regex, in addition Authorized addresses.

**Expected behavior**
Responses are accepted somehow.

**Comment by nguyenkims** [comment]
@Addvilz SL was designed to work with a personal email provider which doesn't change the return path for every email. We use the return path for the authentication check. A workaround would be to disable this check completely but we'd recommend against that.

**Comment by Addvilz** [comment]
@nguyenkims yes, this is a super edge case that only affects people who have their outgoing mail handled by SES or those who use, for example, AWS WorkMail for their personal inboxes.

The best solution I could come up with was to allow regex-based Authorized addresses to have at least some layer of security.

**Comment by muhlba91** [comment]
@Addvilz did you have any success and, if so, care to share your configs/steps to get it working?

i‘m considering the same use case with leveraging AWS SES and require the functionality to reply to (and possibly also send through) aliases.

**Comment by Addvilz** [comment]
@muhlba91, unfortunately, I was not able to make it work.

**Comment by muhlba91** [comment]
@nguyenkims i was wondering if there‘s some sort of documentation about this return path requirement?

i have the following scenario and wasn‘t able to find out when the return path really matters.
i have one mail server for simplelogin domains and the personal domain; hence, incoming mails to a simplelogin alias will be sent by the mail server to the handler which then rewrites the recipient and submits it back to the mail server.
now, the server won‘t change any mail data and just forward it to the local MTA storing the mail => no change to the return path in this case. so far, this should work, right?

now to the case when i reply to a simplelogin rewritten email. replying to an email will work because of the return path; simplelogin translates the recipient to the actual one, auth works, and sends out the email through the mail server.
but in this case, when using an external mail relay like SES the return path wouldn‘t be conserved and is lost when it reaches the original recipient again.
does this create a problem or will it work even though the return path got rewritten for this direction?

**Comment by nguyenkims** [comment]
Hi the return path is used here https://github.com/simple-login/app/blob/disable-pgp-onboarding-proton-mail/email_handler.py#L2425

It's used for checking email authentication. It shouldn't apply to self host as it requires rspamd or spamassassin to work.
